python/triton_kernels/tests/test_matmul.py

```diff
@@ -297,7 +297,12 @@ def test_op(m, n, k, split_k, do_gather, do_scatter, fused_scatter, has_y_gammas
         pytest.skip("fused scatter scratchpad not supported with split_k")
     if hbm_swizzling:
         if is_hip():
-            pytest.skip("NYI. HBM swizzling just implemented for CUDA.")
+            if not is_hip_cdna4():
+                pytest.skip("Scale preshuffling on AMD GPU has not been emulated on non-CDNA4 arch yet.")
+            if "mx" not in weight_dtype_str:
+                pytest.skip("None-scale swizzling not supported on CDNA4 yet")
+            if n % 32 != 0 or k % (32 * 8) != 0:
```

Comment:
We also have restriction on `m` that must be at least 32?

Reply:
In these tests we only preshuffle the scale of weight so we only need the weight scale tensor to be large enough. m>=32, n>=32, k>=256 is the limit on block size, which is satisfied by the constraints on opt_flags.